### ISSUE TITLE
Remove https restriction for notifications

### ISSUE THREAD

**Issue by jonaskuske** [axllent/mailpit#112]
Notifications should also work on localhost, but since mailpit doesn't rely on the browser to check whether notifications are supported or not but instead does its own https verification, the option does not show up in the UI. Would be nice to remove this artifical restriction, happy to contribute! :)

**Comment by axllent** [comment]
Hi @jonaskuske. Actually the HTTPS requirement for notifications is a [browser restriction](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API), not a Mailpit one. I simply detect the HTTPS state to show/hide the subscription button (to the notifications) to prevent user confusion and browser warnings.

Where did you get your "should also work on localhost" information from?

**Comment by jonaskuske** [comment]
On MDN itself! The page you linked has a link to [`secure contexts`](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) right at the top, which says this:

![image](https://github.com/axllent/mailpit/assets/30421456/c07bf07f-d0f0-4017-ba6b-639d774d5101)

> to show/hide the subscription button (to the notifications) to prevent user confusion and browser warnings

Makes absolute sense! I thought this would work just the same without the https check, as `Notification.permission === "denied"` on insecure contexts. However, turns out this is only the case on Chrome and Safari, in Firefox it is `"default"`, so permission state isn't enough to determine support there. You can replace the `https` check with the well-supported property `isSecureContext`, though:

```js
this.notificationsSupported = window.isSecureContext
  && ("Notification" in window && Notification.permission !== "denied");
```

**Comment by axllent** [comment]
You're absolutely correct, I did not realise `localhost` was exempt from HTTPS! Thanks, I've merged your PR. Now I need to make a couple of minor adjustments to related things (including URl displayed in the CLI and in-browser new mail notifications). I appreciate the info and effort, thank you.

**Comment by axllent** [comment]
I have just released this as part of v1.6.11 - sorry it took so long, I had a very busy week!
